lib/model/QubitSetting.php

```diff
@@ -84,13 +84,21 @@ public static function getSettingsArray()
       LEFT JOIN '.QubitSettingI18n::TABLE_NAME.' source
         ON (setting.ID = source.id AND source.CULTURE = setting.SOURCE_CULTURE)';
 
-    $settings = array();
+    $settings = $i18nLanguages = array();
     $culture = sfContext::getInstance()->user->getCulture();
 
     foreach (QubitPdo::fetchAll($sql, array($culture)) as $qubitSetting)
     {
       if ($qubitSetting->scope)
```

Comment:
This looks like compensation for some bad data modelling to me.  @jraddaoui do you think it's practical to change the i18n language setting data from this:

```
mysql> select * from setting where scope='i18n_languages';
+------+----------------+----------+------------+----------------+----+---------------+
| name | scope          | editable | deleteable | source_culture | id | serial_number |
+------+----------------+----------+------------+----------------+----+---------------+
| en   | i18n_languages |        1 |          0 | en             | 27 |             0 |
| fr   | i18n_languages |        1 |          1 | en             | 28 |             0 |
| es   | i18n_languages |        1 |          1 | en             | 29 |             0 |
| nl   | i18n_languages |        1 |          1 | en             | 30 |             0 |
| pt   | i18n_languages |        1 |          1 | en             | 31 |             0 |
| fi   | i18n_languages |        1 |          1 | en             | 48 |             0 |
| uk   | i18n_languages |        1 |          1 | en             | 49 |             0 |
| sk   | i18n_languages |        1 |          1 | en             | 94 |             0 |
+------+----------------+----------+------------+----------------+----+---------------+
8 rows in set (0.00 sec)

mysql> select * from setting_i18n where id in (select id from setting where scope='i18n_languages');
+-------+----+---------+
| value | id | culture |
+-------+----+---------+
| en    | 27 | en      |
| fr    | 28 | en      |
| es    | 29 | en      |
| nl    | 30 | en      |
| pt    | 31 | en      |
| fi    | 48 | en      |
| uk    | 49 | en      |
| sk    | 94 | en      |
+-------+----+---------+
8 rows in set (0.00 sec)
```

to something sensible like:

```
mysql> select * from setting where name='i18n_languages';
+----------------+-------+----------+------------+----------------+----+---------------+
| name           | scope | editable | deleteable | source_culture | id | serial_number |
+----------------+-------+----------+------------+----------------+----+---------------+
| i18n_languages | null  |        1 |          0 | en             | 27 |             0 |
+----------------+-------+----------+------------+----------------+----+---------------+
1 rows in set (0.00 sec)

mysql> select * from setting_i18n where id =27;
+-------------------------+----+---------+
| value                   | id | culture |
+-------------------------+----+---------+
| en,es,fi,fr,nl,pt,sk,uk | 27 | en      |
+-------------------------+----+---------+
1 rows in set (0.00 sec)
```

Or does changing the data fixture require too much work?

Reply:
The i18n_language `value` could be a serialized PHP array if that it easier and/or more resistant to bad data.

Reply:
Thanks @djjuhasz!

That's a great idea but, as you say, I think it requires too much work (fixtures, migration, form, etc.?) for the benefits it will provide (AFAIK). It'd be clearer if we go that way but it won't affect a lot the sfConfig population (as we already loop over all settings) and we'll access the value in the same manner.

Please, let me know if you strongly think we should do it and I'll give it a try ;)

Reply:
@jraddaoui you are probably right that the benefits of fixing the data fixtures are not enough to justify the the effort required. 😦  I would personally *like* to see the data fixed, but I don't have the authority to make that call.